`azurerm_cognitive_account` - enhancement to support kind 'AIServices' (#30423)
hashicorp/terraform-provider-azurerm: website/docs/r/cognitive_account.html.markdown

Reviewer: Unfortunately, this is the standard pattern across all providers for the Registry. It's fine to add a `**Note:**` with a "for more information, please see [LINK]" if there's a goo Learn or Product page to refer to?

Author: @deeikele , do we have any such page to add? We can always at it later.

-> **Note:** SKU `DC0` is the commitment tier for Cognitive Services containers running in disconnected environments. You must obtain approval from Microsoft by submitting the [request form](https://aka.ms/csdisconnectedcontainers) first, before you can use this SKU. More information on [Purchase a commitment plan to use containers in disconnected environments](https://learn.microsoft.com/en-us/azure/cognitive-services/containers/disconnected-containers?tabs=stt#purchase-a-commitment-plan-to-use-containers-in-disconnected-environments).

> * `custom_subdomain_name` - (Optional) The subdomain name used for token-based authentication. This attribute is required when `network_acls` is specified. This attribute is also required when using the OpenAI service with libraries which assume the Azure OpenAI endpoint is a subdomain on `https://openai.azure.com/`, eg. `https://<custom_subdomain_name>.openai.azure.com/`. Once set, it cannot be changed.

Reviewer 2: Currently, the `attribute` and `property` are used interchangably in this document set. Whilst, please at least keep the last sentence as it was, which is a fixed tone to describe force new attributes.

> * `custom_subdomain_name` - (Optional) The subdomain name used for token-based authentication. This attribute is required when `network_acls` is specified. This attribute is also required when using the OpenAI service with libraries which assume the Azure OpenAI endpoint is a subdomain on `https://openai.azure.com/`, eg. `https://<custom_subdomain_name>.openai.azure.com/`. This can be specified during creation or added later, but once set it cannot be changed or removed without forcing a new resource.

Author: Added Entra ID to the first sentence.
`The subdomain name used for Entra ID token-based authentication.`

Given this parameters is technically optional, I don't think it's appropriate to mention it here. It should be part of service documentation.

> -> **Note:** If you do not specify a `custom_subdomain_name` then you will not be able to attach a Private Endpoint to the resource.

Author: Again, we are not checking this in TF. This should go in service documentation. In fact, the documentation doesn't mention 'Azure AI Foundry' at all.

* `custom_subdomain_name` - (Optional) The subdomain name used for token-based authentication. This property is required when `network_acls` is specified. This property is also required when using the OpenAI service with libraries which assume the Azure OpenAI endpoint is a subdomain on `https://openai.azure.com/`, eg. `https://<custom_subdomain_name>.openai.azure.com/`.  Changing this forces a new resource to be created.

> * `dynamic_throttling_enabled` - (Optional) Whether to enable the dynamic throttling for this Cognitive Service Account. This attribute can't be set when kind is `OpenAI` or `AIServices`.

Reviewer 2: ```suggestion
* `dynamic_throttling_enabled` - (Optional) Whether to enable the dynamic throttling for this Cognitive Service Account. This attribute can't be set when the `kind` is `OpenAI` or `AIServices`.
```

> * `dynamic_throttling_enabled` - (Optional) Whether to enable the dynamic throttling for this Cognitive Service Account. This attribute can't be set when kind is `OpenAI` or `AIServices`.

Reviewer 2: This is missing in the schema definition?

Author: It's there on L346-348:

```
			if d.Get("dynamic_throttling_enabled").(bool) && utils.SliceContainsValue([]string{"OpenAI", "AIServices"}, kind) {
				return fmt.Errorf("`dynamic_throttling_enabled` is currently not supported when kind is set to `OpenAI` or `AIServices`")
			}
```

Reviewer 2: I mean it is not defined in the schema.

---

A `network_injection` block supports the following:

> * `scenario` - (Required) Specifies what features network injection applies to. Only possible value is `agent` for agent scenarios.

Reviewer: ```suggestion
* `scenario` - (Required) Specifies what features network injection applies to. The only possible value is `agent`.
```
